Algorithms/Computational-Geometry/arrayRotation.c

```diff
@@ -0,0 +1,62 @@
+//Name = Rishabh Garg
+//E-mail = rishabhgarg25699@gmail.com
+//******* PROGRAM TO ROTATE AN ARRAY OF SIZE N BY ROTATIONS ELEMENTS *********
+//********************* Program in C Language ********************************
+//This program can be done by thee methods with different time and space complexities. 
+//This program is in the best complexity of all the programs.This is done with the help of GCD.
+
+#include <stdio.h>
+int gcd(int , int );
+int main()
+{
+	int arr[50],rotations,n;
+	int i, j, k, temp;
+	printf("Enter how many elements to be stored in an array\n");
+	scanf("%d",&n);
+	printf("Now enter all the elements in the array\n");
+	for(i = 0; i < n; i++)
+	{
+		scanf("%d",&arr[i]);
+	}
+	printf("Enter how much do you want to left rotate the array\n ");
+	scanf("%d",&rotations);
+	//The below function has the main logic of the program.In this we have to find GCD.
+	//The purpose of finding of GCD is that when we got the n and rotations, then we calculate the 
+	//GCD of the both the numbers.Then that array will broke into groups of size no. equal to GCD.
+	//This GCD helps to minimize the comparison within the array.The temination condition came when
+	//GCD goes down to 1.If GCD is one then elements will be moved within their small patch.
+	//With the help of GCD, elements jumps directily leaving rotation variable in one loop.
+	//By this way also,this method decreases the time complexity also. 
+	for (i = 0; i < gcd(rotations, n); i++)       

```

Comment:
This part is directly copied from [geeksforgeeks](https://www.geeksforgeeks.org/array-rotation/). Closing the PR.

Reply:
No,I did not copied this code from anywhere